scripts/webpack-utils/BuildCompiler.js

```diff
@@ -0,0 +1,233 @@
+   *
+   * @returns {void}
+   */
+  initialize () {
+    // @todo - there are many documented hooks, but not all of them appear to be
+    // implemented.  I tried to implement the `beforeRun` hook
+    this.compiler.hooks.run.tap(this.name, () => {
+      console.log(`${this.name}:build:run`);
+
+      this.clearTimer();
+      this.setTimer();
+    });
+
+    // Called when a compilation has finished
+    this.compiler.hooks.done.tap(this.name, (stats) => {
+      console.log(`${this.name}:build:done`);
+
+      this.clearTimer();
+
+      const info = stats.toJson();
+
+      // Show ALL compile results
+      console.log('');
+      console.log(stats.toString({
+        colors: true,
+      }));
+      console.log('');
+
+      // Handle compile warnings
+      if (stats.hasWarnings()) {
+        console.warn('');
+        console.warn('Warnings ----------------------------------------------');
+        console.warn('');
+
+        for (let i = 0; i < info.warnings.length; i++) {
+          console.warn(`Warning #${i + 1}:`);
+          console.warn(chalk.yellowBright(info.warnings[i]));
+          console.warn('');
+        }
+
+        console.warn('');
+        console.warn('^^ Warnings -------------------------------------------');
+        console.warn('');
+      }
+
+      // Handle compile errors
+      if (stats.hasErrors()) {
+        console.error('Errors ------------------------------------------------');
+        console.error('');
+
+        for (let i = 0; i < info.errors.length; i++) {
+          console.warn(`Error #${i + 1}:`);
+          console.warn(chalk.redBright(info.errors[i]));
+          console.warn('');
+        }
+
+        console.error('');
+        console.error('^^ Errors ---------------------------------------------');
+        console.error('');
+
+        console.error('');
+        console.error('Error while compiling with webpack!');
+        console.error(`Webpack compilation finished with ${info.errors.length} errors and ${info.warnings.length} warnings...`);
+        console.error('Make a change and save it to recompile...');
+        console.error('');
+
+        return;
+      }
+
+      console.log('');
+      console.log('Webpack compiled successfully!');
+      console.log(`Webpack compilation finished with ${info.errors.length} errors and ${info.warnings.length} warnings...`);
+      console.log('Make a change and save it to recompile...');
+      console.log('');
+    });
+  }
+
+  /**
+   * @private
+   *
+   * Properly handle a webpack compiler error.  Logs the useful error info to
+   * the console.
+   *
+   * @see - https://webpack.js.org/api/node/#error-handling
+   *
+   * @param {Error} error
+   *   The webpack compiler error
+   */
+  handleBuildError (error) {
+    if (!error) {
+      return;
+    }
+
+    console.error(error.stack || error);
+
+    if (error.details) {
+      console.error(error.details);
+    }
+
+    this.clearTimer();
+
+    return;
+  }
+
+  /**
+   * @async
```

Comment:
@brainthinks per https://jsdoc.app/tags-async.html this is NOT @async. Please update and review your other jsdocs references as you did this for other promise returning methods.

Reply:
done